### PR TITLE
Reg 47/course blocks

### PR DESCRIPTION
**Pull request by arianaolson419**
https://olinjs.atlassian.net/browse/REG-47
https://olinjs.atlassian.net/browse/REG-50 (These two ended up being the same thing)
Planned Courses can be dragged and placed into different semesters. The semester of the course is updated in the state when the course is moved to a new semester.

image

image

### CODE REVIEW

**Comment by arianaolson419** [comment]
The second screenshot is of me dragging the course.

File: src/reducers/LoginReducer.jsx [hunk 1]
```diff
@@ -31,6 +31,17 @@ const LoginReducer = (state = initialState, action) => {
...
+      const isCourseMatch = course => course.courseId === action.courseId;
+      const index = state.plannedCourses.findIndex(isCourseMatch);
+      const course = state.plannedCourses.find(isCourseMatch);
+      course.semester = action.newSemester;
+      const updatedPlanned = state.plannedCourses;
+      updatedPlanned[index] = course;
+      return Object.assign({}, state, {
+        plannedCourses: updatedPlanned,
+      });
+    }
```
Comment (arianaolson419) [inline src/reducers/LoginReducer.jsx:44]: So... the state that I need is in LoginReducer, which doesn't really make sense for actions that have nothing to do with logging in... Would it be better to split up receiving the state at login into a different reducer, or to just change the name of the file to encompass all the actions it handles?

**Comment by robertsiegel** [comment]
Only wondering if we should change around the state so that it can be in a different reducer or rename LoginReducer so that it makes more sense to have CHANGE_SEMESTER in there. Other than that it looks good!

**Review by robbie-sundstrom — APPROVED** [review]
Other than Ariana's comment about the LoginReducer, I think everything looks good!

File: src/reducers/LoginReducer.jsx (hunk 1 above)
Comment (billmwong) [reply src/reducers/LoginReducer.jsx:44]: I'm confused, isn't the Redux state global?  So you should be able to access `state.plannedCourses` from any reducer, and then separating out this `CHANGE_SEMESTER` reducer case should be easy.

File: src/containers/SemesterReserveContainer.jsx [hunk 2]
```diff
@@ -1,22 +1,50 @@
...
-  );
+  ).sort(compare);
   return filteredPlanned;
 };
 
-const mapStateToProps = (state, ownProps) => (
+// How the target reacts to drag and drop events
+const semesterTarget = {
+  drop(props, monitor) {
+    // return drop result available to the drag sourc
```
Comment (billmwong) [inline src/containers/SemesterReserveContainer.jsx]: I think you're missing an 'e' in 'source'

**Review by billmwong — CHANGES_REQUESTED** [review]
Damn Ariana, this code is awesome.  Two comments.

File: src/containers/SemesterReserveContainer.jsx (hunk 2 above)
Comment (arianaolson419) [reply src/containers/SemesterReserveContainer.jsx]: Not the way the state is structured now as far as I can tell... We were combining reducers in the wrong way. I'm working on improving that on another card. I'll fix the 'e'!

**Review by billmwong — APPROVED** [review]
Cool, if that combining reducers bug turns out to be too much, write it on a future ticket for reference and skip it!

**Pull request merged**